src/crosswordgenerator.ts

```diff
@@ -16,138 +17,176 @@ const emptyTile: tileCrossWord = {
 export async function generateCrossword(questions: question[]): Promise<tileCrossWord[][]>{
     let wordCount = questions.length;
     let answers: answer[] = [];
-    for(let i = 0; i < wordCount; i++){
+    let currentAnswers: answer[] = [];
+    questions.forEach((question,index) => {
         answers.push({
-            answer: questions[i].answer,
-            questNumber: i+1,
+            answer: question.answer,
+            questNumber: index+1,
         })
-    }
-
-    let crossword: tileCrossWord[][] = await simpleCrossWord(answers);
+        currentAnswers.push({
+            answer: question.answer,
+            questNumber: index+1,
+        })
+    })
+    let crossword: tileCrossWord[][] = await simpleCrossWord(answers, questions);
     let score: number = await getScore(crossword);
 
-    for(let i = 0; i<4; i++){
-        let currentCrossword: tileCrossWord[][] = await simpleCrossWord(answers);
+    for(let i = 0; i<10  ; i++){
```

Comment:
If you want to you can increase the number of iterations to get the same count of crossword-generations

Reply:
You are right. I fixed it.